src/storage/txn/txn_status_cache.rs

```diff
@@ -104,7 +104,32 @@
 //! lock; and for write requests, if it finds the transaction of that lock is
 //! already committed, it can merge together the resolve-lock-committing and the
 //! write operation that the request needs to perform.
-
+//!
+//! ## Design Update-1: Dual-Cache System
+//!
+//! Goal: maximize all TiKV nodes' awareness of large pipelined transactions
+//! during their lifetime, i.e. from their first writes to all locks being
+//! committed. This is crucial to resolved-ts resolver to handle locks belonging
+//! to large transactions.
+//!
+//! The txn_status_cache is then split into two independent parts.
+//! 1. `normal_cache`: For most transactions, including committed and rolled
+//!    back transactions.
+//! 2. `large_txn_cache`: Specifically for ongoing large transactions.
+//!
+//! ### Key characteristics:
+//!
+//! - Large Transaction Identification: Large transactions are identified if
+//!   their `start_ts` and `min_commit_ts` differ. Non-large transactions which
+//!   request to be cached in the cache can also be treated as large
+//!   transactions, as they imply their min_commit_ts are useful.
+//!
+//! - Prioritized Caching: The `large_txn_cache` has a higher priority when
```

Comment:
Perhaps we should make it clear the `large transactions` here indicates only pipelined dml transactions.

Reply:
While the project currently focuses on pipelined transactions, the module itself doesn't need to be transaction-type specific. It can handle caching `min_commit_ts` for any large transaction that requires it, not just pipelined DMLs. This approach offers more flexibility and future-proofs our design.